### PR TITLE
RevDiff: Setting to show Blame in FileTree

### PR DESCRIPTION
**Pull request by gerhardol**
Part of #9796 

This addresses what is maybe the core problem in #9796: Blame in RevDiff
This was added in #9424 and was intended for quick checks. The use
corresponding to FileHistory was intended to be used in FileTree.

(Some other changes planned for Blame in general and specifically for Blame in RevDiff for other concerns.)

## Proposed changes

Blame in RevDiff has certain limitations as revisions for a file may
not exist in the grid and the file to blame may not exist in all
revisions.

Add a setting to explicitly enable blame in RevGrid.

### Before

Blame 'B' in RevDiff by default opened in RevDiff itself

image

### After

Default is only FileTree

image

Blame 'B' in RevDiff by default opens FileTree
Basically the same as selecting filetree in the context menu 'T' but Blame is opened if show file contents was selected ('B' in FileTree).

image

## Test methodology 

Manual

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommandsDialogs/RevisionDiffControl.cs [hunk 1]
```diff
@@ -476,10 +476,8 @@ private void ResetSelectedItemsTo(bool actsAsChild)
             RequestRefresh();
         }
 
-        private void BlameSelectedFileDiff()
+        private void BlameSelectedFileDiff(int? line = null)
         {
-            int? line = DiffText.Visible ? DiffText.CurrentFileLine : null;
```
Comment (mstv) [inline GitUI/CommandsDialogs/RevisionDiffControl.cs:481]: Sure? This argument is never used.

Comment (gerhardol) [reply GitUI/CommandsDialogs/RevisionDiffControl.cs:481]: fixed the caller
Too many conflicting changes right now...

Comment (mstv) [reply GitUI/CommandsDialogs/RevisionDiffControl.cs:481]: > Too many conflicting changes right now...

Be patient, please. You have thought through a lot and have integrated the history and blame functionality pretty smoothly IMO. I was afraid it would become more intrusive / complex.
I don't think it will be too difficult to support the other users' expectations step by step.

> fixed the caller

partially: the call yet

Comment (gerhardol) [reply GitUI/CommandsDialogs/RevisionDiffControl.cs:481]: hopefully done...
I have some difficult merges to do...

**Review by mstv — APPROVED** [review]

**Review by RussKie — COMMENTED** [review]
:shipit:

**Comment by mstv** [comment]
For the artificial commits (where FileTree is not available), the Blame action should not just be disabled. Then I'd rather show the Blame in the Diff tab.

**Comment by gerhardol** [comment]
> For the artificial commits (where FileTree is not available), the Blame action should not just be disabled. Then I'd rather show the Blame in the Diff tab.

That would be inconsistent behavior, slightly confusing.

I would have liked to switch to HEAD and showed it there (that is what Blame in artificial does anyway), but that was not so easy.
